src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -5141,6 +5184,234 @@ static synchronized long getColumnEncryptionKeyCacheTtl() {
+     */
+    static public int getDefaultServerPreparedStatementDiscardThreshold() {
+        if(0 > defaultServerPreparedStatementDiscardThreshold)
+            return INITIAL_DEFAULT_SERVER_PREPARED_STATEMENT_DISCARD_THRESHOLD;
+        else
+            return defaultServerPreparedStatementDiscardThreshold;        
+    }
+
+    /**
+     * Specifies the default behavior for new connection instances. This setting controls how many outstanding prepared statement discard
+     * actions (sp_unprepare) can be outstanding per connection before a call to clean-up the outstanding handles on the server is executed.
+     * If the setting is <= 1 unprepare actions will be executed immedietely on prepared statement close. If it is set to >1 these calls will
+     * be batched together to avoid overhead of calling sp_unprepare too often. 
+     * Initial setting for this option is available in INITIAL_DEFAULT_SERVER_PREPARED_STATEMENT_DISCARD_THRESHOLD.
+     * 
+     * @param value
+     *      Changes the setting per the description.
+     */
+    static public void setDefaultServerPreparedStatementDiscardThreshold(int value) {
+        defaultServerPreparedStatementDiscardThreshold = value; 
+    }
+
+    /**
+     * Returns the behavior for a specific connection instance. This setting controls how many outstanding prepared statement discard
+     * actions (sp_unprepare) can be outstanding per connection before a call to clean-up the outstanding handles on the server is executed.
+     * If the setting is <= 1 unprepare actions will be executed immedietely on prepared statement close. If it is set to >1 these calls will
+     * be batched together to avoid overhead of calling sp_unprepare too often. 
+     * The default for this option can be changed by calling getDefaultServerPreparedStatementDiscardThreshold(). 
+     * 
+     * @return Returns the current setting per the description.
+     */
+    public int getServerPreparedStatementDiscardThreshold() {
+        if(0 > this.serverPreparedStatementDiscardThreshold)
+            return getDefaultServerPreparedStatementDiscardThreshold();
+        else
+            return this.serverPreparedStatementDiscardThreshold;        
+    }
+
+    /**
+     * Specifies the behavior for a specific connection instance. This setting controls how many outstanding prepared statement discard
+     * actions (sp_unprepare) can be outstanding per connection before a call to clean-up the outstanding handles on the server is executed.
+     * If the setting is <= 1 unprepare actions will be executed immedietely on prepared statement close. If it is set to >1 these calls will
+     * be batched together to avoid overhead of calling sp_unprepare too often. 
+     * 
+     * @param value
+     *      Changes the setting per the description.
+     */
+    public void setServerPreparedStatementDiscardThreshold(int value) {
+        this.serverPreparedStatementDiscardThreshold = value;
```

Comment:
Nit Pick: In the code you have deferenced all the member vars and methods using `this` keyword. Looks like the files convention is to use the member or method directly. 

I leave this to the committers to state what the guideline is for JDBC.